src/policy/compiler.rs

```diff
@@ -992,8 +992,23 @@ where
                     }
                 })
                 .collect();
-            if key_vec.len() == subs.len() && subs.len() <= 20 {
-                insert_wrap!(AstElemExt::terminal(Terminal::Multi(k, key_vec)));
+            if key_vec.len() == subs.len() {
+                if subs.len() <= 20 {
+                    insert_wrap!(AstElemExt::terminal(Terminal::Multi(k, key_vec)));
+                } else if subs.len() == k {
+                    // If this is a N-of-N, then we can save some bytes in the witness
+                    // program by just CHECKSIGVERIFY on all keys instead of counting.
+                    let mut policy = Concrete::And(vec![
```

Comment:
I think this is not special to keys, The else if code can be moved to match thresh rule. Transforming the 

Concrete::thresh(k, ref subs) into chain of `Ands` if `k == n`. That way this will also work for things that are not keys and for n < 20 too.

Reply:
Hmm, I'm not sure it's efficient for `n < 20` ?

If using `CHECKMULTISIG` you have `4` bytes taken by OP codes (`n` pushed 2 times, `CHECKMULTISIG`, and the empty byte array), while you have `n` bytes taken by OP codes if using a set of `CHECKSIGVERIFY`s.

So, yeah, it's worth it for `n < 4 || n > 20`, will change that.

Reply:
I mean the there can be other `subs` apart from keys that can benefit from chaining `And` compilation. It is most likely the case that `thresh(1,sha256(H),pk(C),pk(C),pk(C))` should be treated as `and(sha256(H),and(pk(C),and(pk(C),pk(C))` would produce a better compilation

Reply:
Ok, so we want this for policies where `k == subs.len()` *but not* when all `subs` are keys and `k in 4..=20` (in which case we want `multi`) ?

Reply:
Finally went for:
- All keys and less than 20 keys ?
  - Yes => `multi`
  - No
    - => `k == subs.len()` ?
      - Yes => Nested `and()`s
      - No => `thresh()`